services/brig/test/integration/API.hs

```diff
@@ -447,6 +450,40 @@ testMultipleUsers brig = do
     field :: FromJSON a => Text -> Value -> Maybe a
     field f u = u ^? key f >>= maybeFromJSON
 
+
+testCreateUserAnonExpiry :: Brig -> Http ()
+testCreateUserAnonExpiry b = do
+    u1 <- randomUser b
+    alice <- randomUser b
+    bob <- createAnonUser "bob" b
+    liftIO $ assertBool "expiry not set on regular creation" (not $ isJust $ userExpire alice)
+    ensureExpiry (userExpire bob) "bob/register"
+    resAlice <- get (b . zUser (userId u1) . paths ["users", toByteString' (userId alice)]) <!! const 200 === statusCode
+    resBob <- get (b . zUser (userId u1) . paths ["users", toByteString' (userId bob)]) <!! const 200 === statusCode
+    selfBob <- get (b . zUser (userId bob) . path "self") <!! const 200 === statusCode
+    unless (null (expire resAlice)) $
+        liftIO $ assertFailure "Regular user should not have any expiry"
+    ensureExpiry (expire resBob) "bob/public"
+    ensureExpiry (expire selfBob) "bob/self"
+  where
+    ensureExpiry :: Maybe UTCTime -> String -> Http ()
+    ensureExpiry expiry s = case expiry of
+                       Nothing -> liftIO $ assertFailure ("user must have an expiry" <> s)
+                       Just a -> do
+                          now <- liftIO getCurrentTime
+                          let diff = diffUTCTime a now
+                              minExp = 20 :: Integer -- 20 seconds
+                              maxExp = 60 * 60 * 24 * 10 :: Integer -- 10 days
+                          liftIO $ assertBool "expiry in the future" (diff > fromIntegral minExp)
```

Comment:
[nit-pick] which way does the text go here?  should it be the one printed in case of failure, or the condition that is tested here?

Reply:
Good point, thanks!